apps/marketplace/components/BuyerBriefFlow/DateControl.js

```diff
@@ -0,0 +1,155 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { Control } from 'react-redux-form'
+import AUtextInput from '@gov.au/text-inputs/lib/js/react'
+import styles from './DateControl.scss'
+
+const parseValue = value => {
+  let date = {
+    day: '',
+    month: '',
+    year: ''
+  }
+  if (value && value.match(/\d{4}-\d{1,2}-\d{1,2}/)) {
+    date = {
+      day: value.split('-')[2] ? value.split('-')[2] : '',
+      month: value.split('-')[1] ? value.split('-')[1] : '',
+      year: value.split('-')[0] ? value.split('-')[0] : ''
+    }
+  }
+  return date
+}
+
+export class DateComponent extends Component {
+  constructor(props) {
+    super(props)
+
+    const { day, month, year } = parseValue(props.value)
+    this.state = {
+      day,
+      month,
+      year
+    }
+
+    this.handleChange = this.handleChange.bind(this)
+  }
+
+  handleChange(e) {
+    const name = e.target.name
+    const value = e.target.value
+    this.setState(curState => {
+      const newState = { ...curState }
+      newState[name] = value
+      this.props.onDateChange(newState)
+      return newState
+    })
+  }
+
+  render() {
+    return (
+      <div className={`row ${this.props.className}`}>
+        <div className="col-xs-12">
+          <label htmlFor="closing_date" className="question-heading au-text-input__label">
+            {this.props.label}
+          </label>
+          <div>{this.props.description}</div>
+        </div>
+        <div id="closing_date" className={`col-xs-12 col-md-6 ${styles.inputsContainer}`}>
+          <div>
+            <label htmlFor="day">Day</label>
+            <AUtextInput
+              block
+              id="day"
+              type="number"
+              name="day"
+              value={this.state.day}
+              onChange={this.handleChange}
+              min="1"
+              max="31"
```

Comment:
Can we alter the max depending on the month value? It looks like we don't create invalid dates when one is submitted but the user can still enter an invalid date without any feedback that something is wrong.

Reply:
fixed